### ISSUE TITLE
Toggle requires aria-label but has no effect

### ISSUE THREAD

**Issue by janhassel** [carbon-design-system/carbon#8432]
## Environment

> Operating system
macOS 11.1

> Browser
Chrome 89

> Automated testing tool and ruleset
n/a

> Assistive technology used to verify
VoiceOver

## Detailed description

> What version of the Carbon Design System are you using?
`carbon-components-react@10.33.0`
`carbon-components@7.33.0`

> What did you expect to happen?
A provided `aria-label` for `Toggle`, which is marked as required, should be picked up by the screen reader.
https://github.com/carbon-design-system/carbon/blob/6544a3b6cad0f2d3272c97d482ab216780eab9a1/packages/react/src/components/Toggle/Toggle.js#L20

> What happened instead?
The `aria-label` is not being picked up. The only way to label a toggle is to pass a `labelText` which is not a required prop. A lot of teams are using the `labelA` and `labelB` props to label their toggles especially when multiple toggles are stacked or they appear in toolbars. However, these two labels are hidden to screen readers which is an additional motivation to pass in an `aria-label`. Sadly, it doesn't get read.

![Screen Recording 2021-04-16 at 10 47 25](https://user-images.githubusercontent.com/28265588/114999773-7339f200-9ea2-11eb-9802-89c68faa081a.gif)

> What WCAG 2.1 checkpoint does the issue violate?
SC 3.3.2

## Steps to reproduce the issue

https://codesandbox.io/s/modest-buck-68eun?file=/src/App.js

## Additional information

How about offering a variant of the toggle component which by default shows the label next to the toggle (like in the second example)? Like mentioned above, teams already do this, but the implementation is likely inaccessible and feels cumbersome (providing the same label three times to the component).
